x-pack/plugins/enterprise_search/server/routes/app_search/credentials.test.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { MockRouter, mockConfig, mockLogger } from '../__mocks__';
+
+import { registerCredentialsRoutes } from './credentials';
+
+jest.mock('./app_search_request_handler', () => ({
+  createAppSearchRequestHandler: jest.fn(),
+}));
+import { createAppSearchRequestHandler } from './app_search_request_handler';
+
+describe('credentials routes', () => {
+  describe('GET /api/app_search/credentials/collection', () => {
+    let mockRouter: MockRouter;
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+      mockRouter = new MockRouter({ method: 'get', payload: 'query' });
+
+      registerCredentialsRoutes({
+        router: mockRouter.router,
+        log: mockLogger,
+        config: mockConfig,
+      });
+    });
+
+    it('create a handler with app_search_request_handler', () => {
+      expect(createAppSearchRequestHandler).toHaveBeenCalledWith({
+        config: mockConfig,
+        log: mockLogger,
+        path: '/as/credentials/collection',
+        hasValidData: expect.any(Function),
+      });
+    });
+
+    describe('hasValidData', () => {
+      it('should correctly validate that a response has data', () => {
+        const response = {
+          meta: {
+            page: {
+              current: 1,
+              total_pages: 1,
+              total_results: 1,
+              size: 25,
+            },
+          },
+          results: [
+            {
+              id: 'loco_moco_account_id:5f3575de2b76ff13405f3155|name:asdfasdf',
+              key: 'search-fe49u2z8d5gvf9s4ekda2ad4',
+              name: 'asdfasdf',
+              type: 'search',
+              access_all_engines: true,
+            },
+          ],
+        };
+
+        const hasValidData = (createAppSearchRequestHandler as jest.Mock).mock.calls[0][0]
+          .hasValidData;
```

Comment:
Hmm, this isn't the easiest on the eyes, I wonder if we could abstract it out somewhere.. but in the meanwhile, you could do this:

```suggestion
        const { hasValidData } = (createAppSearchRequestHandler as jest.Mock).mock.calls[0][0];
```